crates/volta-core/src/tool/package/mod.rs

```diff
@@ -182,20 +190,17 @@ fn remove_config_and_shim(bin_name: &str, pkg_name: &str) -> Fallible<()> {
 /// all the binaries installed by the input package.
 fn binaries_from_package(package: &str) -> Fallible<Vec<String>> {
     let bin_config_dir = volta_home()?.default_bin_dir();
-    if bin_config_dir.exists() {
-        dir_entry_match(&bin_config_dir, |entry| {
-            let path = entry.path();
-            if let Ok(config) = BinConfig::from_file(path) {
-                if config.package == package {
-                    return Some(config.name);
-                }
-            };
-            None
-        })
-        .with_context(|| ErrorKind::ReadBinConfigDirError {
-            dir: bin_config_dir.to_owned(),
-        })
-    } else {
-        Ok(vec![])
-    }
+    dir_entry_match(&bin_config_dir, |entry| {
+        let path = entry.path();
+        if let Ok(config) = BinConfig::from_file(path) {
+            if config.package == package {
+                return Some(config.name);
+            }
+        };
+        None
+    })
+    .with_context(|| ErrorKind::ReadBinConfigDirError {
+        dir: bin_config_dir.to_owned(),
+    })
+    .accept_error_as_default_if(|err| err.is_not_found_error_kind())
```

Comment:
Since `dir_entry_match` actually returns an `io::Error`, I think this could be simplified to an `or_else` call on the original Result (_before_ calling `with_context`). We actually already have a (currently private) function in `volta_core::fs` named `ok_if_not_found`, which is set up to work for unit types only, but could be made generic over any type that implements `Default`.

Reply:
Done